packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -436,12 +460,29 @@ export class SharedTreeFork implements ISharedTreeFork {
 	}
 
 	public readonly transaction: ISharedTreeView["transaction"] = {
-		start: () => this.branch.startTransaction(new ForestRepairDataStore(() => this.forest)),
+		start: () => {
+			// A commit repair data store is only needed if there is not already an ongoing transaction.
+			if (this.branch.isTransacting()) {
+				return this.branch.startTransaction(new ForestRepairDataStore(() => this.forest));
+			}
+
+			const clonedForest = this.forest.clone(this.storedSchema.clone(), new AnchorSet());
+			return this.branch.startTransaction(
+				new ForestRepairDataStore(() => this.forest),
+				// Create a RepairDataStore with a clone of the forest, so that repair data can be stored for
+				// the resulting commit rather than just for each change within the transaction.
+				new ForestRepairDataStore(() => clonedForest),
+			);
```

Comment:
It would be nice to dedup this code with the one in SharedTree. Noah might have some thoughts on how to best do that.

Reply:
This code was removed with the RepairDataStoreProvider changes.